### PR TITLE
feat: add warning in dev mode when method is not function (#…

### PR DESCRIPTION
**Pull request by yaodingyd**
#8017
add warning in dev mode when method is not function 

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/core/instance/state.js [hunk 1]
```diff
@@ -273,6 +273,11 @@ function initMethods (vm: Component, methods: Object) {
           `Avoid defining component methods that start with _ or $.`
         )
       }
+      if (typeof methods[key] !== 'function') {
```
Comment (posva) [inline src/core/instance/state.js]: Could you make this test in development mode only and add a test (or/and adapt existing one), please?

File: src/core/instance/state.js [hunk 2]
```diff
@@ -273,6 +273,12 @@ function initMethods (vm: Component, methods: Object) {
           `Avoid defining component methods that start with _ or $.`
         )
       }
+      if (methods[key] != null && typeof methods[key] !== 'function') {
+        warn(
+          `Method "${key}" should be a function.`
+        )
+        break
```
Comment (posva) [inline src/core/instance/state.js]: you could instead set the method[key] to null, so the code works but does nothing

File: src/core/instance/state.js [hunk 3]
```diff
@@ -273,6 +273,12 @@ function initMethods (vm: Component, methods: Object) {
           `Avoid defining component methods that start with _ or $.`
         )
       }
+      if (methods[key] != null && typeof methods[key] !== 'function') {
```
Comment (posva) [inline src/core/instance/state.js]: instead of checking if it's null, move it up with an `else if`

**Comment by sirlancelot** [comment]
A check and a warning already exists for this in the block directly above the one added by this PR.

**Comment by yaodingyd** [comment]
that's sad.

**Pull request closed without merging**